### PR TITLE
fix(ListSection): Fixing accessibility and spacing

### PR DESCRIPTION
**Pull request by ZackMFleischman**
### Overview

There were 2 problems that came up in the `ListSection`:
1. ListSection wasn't handling accessibility well (no `list` role), and passing a `<ul>` element as the child of it would interpret the entire list as a single child and break the "show all" functionality.

2. In addition, we were in-lining the children in the same container as the header and the footer, so if the consumer wanted to add padding between the elements but _not_ between the last element and the footer or the first element and the header, they were unable to in CSS, having to fallback on a javascript solution checking the indices and manually overriding the padding if it's the first/last element in the list. 

Both of these problems are addressed in this PR with the following changes:
1. The ListSection now renders the body inside of an `ul` and wraps each child in a `li` element. This way we get accessibility for free and the consumer doesn't have to think about it.
2. We automatically apply a default margin between the list items so the consumer doesn't have to think about padding them out. This padding can be overridden with the optional prop `listItemGap`. 

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: https://codecademy.atlassian.net/browse/EGG-1080
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut-labs/src/ListSection/index.tsx [hunk 1]
```diff
@@ -13,29 +15,54 @@ export type ListSectionProps = {
...
 };
 
+const UnstyledUnorderedList = styled.ul`
+  list-style: none;
+  padding: 0;
+  margin: 0;
+`;
+
+type StyledListItemProps = {
+  listItemGap?: number;
```
Comment (ZackMFleischman) [inline packages/gamut-labs/src/ListSection/index.tsx:31]: I'd ideally like this to not be a number, but a system prop value. I'm not sure how to type that though?

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.66-alpha.bd4fb1.0<br/>@codecademy/gamut-labs@15.2.6-alpha.bd4fb1.0<br/>@codecademy/styleguide@36.9.3-alpha.bd4fb1.0<br/></pre>

File: packages/gamut-labs/src/ListSection/index.tsx [hunk 2]
```diff
@@ -1,41 +1,68 @@
...
    * @remarks
    * If the initial display amount matches the number of list items, Show All button will not appear.
    */
   initialDisplayAmount?: number;
-  headerButton?: SectionButton;
-  headerSecondaryButton?: SectionButton;
+  /**
+   * The spacing between list items. Defaults to 24.
+   */
+  listItemGap?: number;
```
Comment (ZackMFleischman) [inline packages/gamut-labs/src/ListSection/index.tsx:21]: I'd like this to not be a number, but be restricted to our normal margin values for our design system but I'm not sure how to type that?

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.66-alpha.c4d756.0<br/>@codecademy/gamut-labs@15.2.6-alpha.c4d756.0<br/>@codecademy/styleguide@36.9.3-alpha.c4d756.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608c3da15c91d800a9e37695--gamut-preview.netlify.app>https://608c3da15c91d800a9e37695--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608c3da15c91d800a9e37695>Deploy Logs</a></p>

File: packages/gamut-labs/src/ListSection/index.tsx (hunk 2 above)
Comment (katiezutter) [reply packages/gamut-labs/src/ListSection/index.tsx:21]: Good question 👀

File: packages/styleguide/stories/Molecules/ListSection.stories.mdx [hunk 3]
```diff
@@ -94,6 +96,41 @@ By default, all of the optional buttons are rendered with the `<TextButton>` com
...
+        >
+          {listItems.map((item) => (
+            <Card>List Item {item.title}</Card>
+          ))}
+        </ListSection>
+      </>
+    )}
+  </Story>
+</Canvas>
+
```
Comment (katiezutter) [inline packages/styleguide/stories/Molecules/ListSection.stories.mdx:133]: 👌

File: packages/gamut-labs/src/ListSection/index.tsx (hunk 2 above)
Comment (ZackMFleischman) [reply packages/gamut-labs/src/ListSection/index.tsx]: We decided to lock this to `24px` and not give the option to tweak it without a design need.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.67-alpha.91c827.0<br/>@codecademy/gamut-labs@15.2.7-alpha.91c827.0<br/>@codecademy/styleguide@36.9.4-alpha.91c827.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608c86f7b42fb02a40535a9f--gamut-preview.netlify.app>https://608c86f7b42fb02a40535a9f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608c86f7b42fb02a40535a9f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.67-alpha.7dc535.0<br/>@codecademy/gamut-labs@15.2.7-alpha.7dc535.0<br/>@codecademy/styleguide@36.9.4-alpha.7dc535.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608c8fc6ccf2782d5bbc465b--gamut-preview.netlify.app>https://608c8fc6ccf2782d5bbc465b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608c8fc6ccf2782d5bbc465b>Deploy Logs</a></p>

**Review by dreamwasp — COMMENTED** [review]
tagging @sooinc !

**Review by JoshuaKGoldberg — APPROVED** [review]
🆒

**Review by dreamwasp — COMMENTED** [review]
nice accessibility upgrade + sounds great on Voiceover, would love to see a monolith integration branch to play around with before i approve

File: packages/gamut-labs/src/ListSection/index.tsx (hunk 2 above)
Comment (katiezutter) [reply packages/gamut-labs/src/ListSection/index.tsx]: Cool makes sense :)

Comment (JoshuaKGoldberg) [reply packages/gamut-labs/src/ListSection/index.tsx]: https://codecademy.slack.com/archives/CQ98K8PT6/p1619800658014300

**Review by katiezutter — APPROVED** [review]
😎

File: packages/gamut-labs/src/ListSection/index.tsx (hunk 2 above)
Comment (katiezutter) [reply packages/gamut-labs/src/ListSection/index.tsx]: 🙏

**Review by sooinc — APPROVED** [review]
LGTM! thanks for making these fixes!

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609168deda25ad35f8d06324--gamut-preview.netlify.app>https://609168deda25ad35f8d06324--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609168deda25ad35f8d06324>Deploy Logs</a></p>

**Pull request merged**